python/ray/serve/scripts.py

```diff
@@ -477,6 +482,9 @@ def build(import_path: str, app_dir: str, output_path: Optional[str]):
         config, Dumper=ServeBuildDumper, default_flow_style=False, sort_keys=False
     )
 
+    # Remove extraneous newline
+    config_str = config_str[:-1]
```

Comment:
is this really necessary? if so, please use `config.rstrip("\n")` so it's more readable

Reply:
Without that line, the file ends with two newlines. Ideally, it should end with a single newline.

I've replaced that snippet with:

```python
# Ensure file ends with only one newline
config_str = config_str.rstrip("\n") + "\n"
```